packages/tasit-identity-contract/src/GnosisSafeUtils.js

```diff
@@ -0,0 +1,149 @@
+import Action from "tasit-action";
+const { Contract } = Action;
+import { ethers } from "ethers";
+
+const { utils: ethersUtils } = ethers;
+const { bigNumberify } = ethersUtils;
+
+// 100 gwei
+const GAS_PRICE = bigNumberify(`${1e11}`);
+
+export default class GnosisSafeUtils {
+  #contract;
+
+  constructor(contract) {
+    this.#contract = contract;
+  }
+
+  // Sign a hash using N signers
+  // Based on: https://github.com/gnosis/safe-contracts/blob/102e632d051650b7c4b0a822123f449beaf95aed/test/utils.js#L93
+  multiSign = (wallets, hash) => {
+    let signatures = "0x";
+
+    wallets.forEach(wallet => {
+      const signingKey = new ethers.utils.SigningKey(wallet.privateKey);
+      const sig = signingKey.signDigest(hash);
+      signatures += ethers.utils.joinSignature(sig).slice(2);
+    });
+
+    return signatures;
+  };
+
+  // Based on: https://github.com/gnosis/safe-contracts/blob/102e632d051650b7c4b0a822123f449beaf95aed/test/utilsPersonalSafe.js#L30
+  estimateFromSafeTxGas = async (to, value, data, operation) => {
```

Comment:
We could use a more general util like this in `tasit-action`

Reply:
That function seems a Safe specific estimation (that use `safeContract.requiredTxGas()`)

Reply:
Agreed. Just a reminder that we might want something similar but more general for other purposes. Resolving this.

Reply:
Or maybe it can't be generalized efficiently and we'll always need specific versions - that's fine too